### ISSUE TITLE
cr.AwsCustomResource: Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1

### ISSUE THREAD

**Issue by svyotov** [aws/aws-cdk#23736]
### Describe the bug

CDK is failing with `Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1`
when we use `cr.AwsCustomResource` with cross account `assumedRoleArn` role.
It works fine when it is called only once, but if there are two separate calls with different accounts assumedRoleArn the resource fails with the error above.

### Expected Behavior

Not to fail

### Current Behavior

> Missing credentials in config, if using AWS_CONFIG_FILE, set AWS_SDK_LOAD_CONFIG=1

### Reproduction Steps

```
const id = "..."
const assumedRoleArn1 = "arn:..."
const assumedRoleArn2 = "arn:..."
const scope = "..."

[[assumedRoleArn1, "first"], [assumedRoleArn2, "second"]].forEach(values => {
  const assumedRoleArn = values[0];
  const parameterName = values[1];
  const resourceId = id + parameterName
  cr.AwsCustomResource(scope, resourceId, {
    resourceType: `Custom::getParameter-${parameterName}`,
    onUpdate:  {
      assumedRoleArn: assumedRoleArn,
      service: 'SSM',
      action: 'getParameter',
      parameters: {
        Name: parameterName
      },
      physicalResourceId: { id: resourceId }
    },
    logRetention: 30,
    functionName: `getParameter-${parameterName}`,
    policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
      resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE
    })
  }).getResponseField('Parameter.Value');
}

```

### Possible Solution

assumedRoleArn1 and assumedRoleArn2 are from 2 different AWS accounts, separate from the account running the cloudformation

### CDK CLI Version

2.60.0 (build 2d40d77)

### Framework Version

om

### Node.js Version

v16.14.0

### OS

Linux

### Language

Typescript

### Language Version

TypeScript (3.8.3)

**Comment by Lewenhaupt** [comment]
I think this is related to https://github.com/aws/aws-cdk/issues/15425

**Comment by pahud** [comment]
resolving from https://github.com/aws/aws-cdk/issues/15425

Feel free to create a new issue if you believe it's still relevant. Thanks.
